### PR TITLE
Feature/docs gen

### PR DESCRIPTION
**Pull request by ferranbt**
This PR creates a new tool to generate docs for the `suave-std` contracts.

The tool is separated in two steps. First, read the artifacts built by `forge` and create a JSON file with the summary, names, fields, descriptions, etc... for every contract with documentation in the artifacts folder. Second, use a simple template engine to generate the documentation in markdown file from the data generated in the previous step.

The idea to separate in two stages is that eventually we might have the summary generated here but the target for the docs (suave-docs) might decide how to render then.

To generate the docs use:
```
$ cd tools/docs-gen
$ go run main.go --suave-std ../../ --out ../../docs
```

As part of this PR I also added some basic documentation for all the major contracts in `suave-std`.

### CODE REVIEW

File: src/suavelib/Suave.sol [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.8;
 
+/// @notice Library to interact with the Suave MEVM precompiles.
```
Comment (ferranbt) [inline src/suavelib/Suave.sol:4]: Manually copied this from https://github.com/flashbots/suave-geth/pull/218

File: src/Transactions.sol [hunk 2]
```diff
@@ -226,6 +287,9 @@ library Transactions {
         return txStruct;
     }
 
+    /// @notice decode a EIP-1559 transaction from RLP.
+    /// @param rlp is the encoded RLP bytes.
+    /// @return txStruct the transaction structure.
```
Comment (zeroXbrock) [inline src/Transactions.sol:292]: Would it be possible to replace `txStruct` with `EIP1559 struct` here?

**Review by zeroXbrock — APPROVED** [review]
this is awesome! just one small suggestion on the return type for the `decodeRLP_...` functions but if it's complicated to implement, then I'm fine with it as is. will leave it to your discretion @ferranbt

File: src/Transactions.sol (hunk 2 above)
Comment (ferranbt) [reply src/Transactions.sol:292]: Mm.. not really, that first word is the name of the return argument and I am not sure I can use EIP1559 as return name.

**Pull request merged**